src/contrib/cluster/Akka.DistributedData/DeltaPropagationSelector.cs

```diff
@@ -55,89 +57,83 @@ public void Delete(string key)
+                foreach (var entry in _deltaEntries)
                 {
-                    // collect the deltas that have not already been sent to the node and merge
-                    // them into a delta group
-                    var deltas = ImmutableDictionary<string, (IReplicatedData, long, long)>.Empty.ToBuilder();
-                    foreach (var entry in _deltaEntries)
-                    {
-                        var key = entry.Key;
-                        var entries = entry.Value;
+                    var key = entry.Key;
+                    var entries = entry.Value;
 
-                        var deltaSentToNodeForKey = _deltaSentToNode.GetValueOrDefault(key, ImmutableDictionary<Address, long>.Empty);
+                    var deltaSentToNodeForKey = _deltaSentToNode.GetValueOrDefault(key, ImmutableDictionary<Address, long>.Empty);
                         
-                        var j = deltaSentToNodeForKey.GetValueOrDefault(node, 0L);
-                        var deltaEntriesAfterJ = DeltaEntriesAfter(entries, j);
-                        if (!deltaEntriesAfterJ.IsEmpty)
-                        {
-                            var fromSeqNr = deltaEntriesAfterJ.Keys.First(); // should be min
-                            var toSeqNr = deltaEntriesAfterJ.Keys.Last(); // should be max
+                    var j = deltaSentToNodeForKey.GetValueOrDefault(node, 0L);
+                    var deltaEntriesAfterJ = DeltaEntriesAfter(entries, j);
+                    if (!deltaEntriesAfterJ.IsEmpty)
+                    {
+                        var fromSeqNr = deltaEntriesAfterJ.Keys.First(); // should be min
+                        var toSeqNr = deltaEntriesAfterJ.Keys.Last(); // should be max
 
-                            // in most cases the delta group merging will be the same for each node,
-                            // so we cache the merged results
-                            var cacheKey = (key, fromSeqNr, toSeqNr);
-                            if (!cache.TryGetValue(cacheKey, out var deltaGroup))
+                        // in most cases the delta group merging will be the same for each node,
+                        // so we cache the merged results
+                        var cacheKey = (key, fromSeqNr, toSeqNr);
+                        if (!cache.TryGetValue(cacheKey, out var deltaGroup))
+                        {
+                            deltaGroup = deltaEntriesAfterJ.Values.Aggregate((d1, d2) =>
                             {
-                                using (var e = deltaEntriesAfterJ.Values.GetEnumerator())
-                                {
-                                    e.MoveNext();
-                                    deltaGroup = e.Current;
-                                    while (e.MoveNext())
-                                    {
-                                        deltaGroup = deltaGroup.Merge(e.Current);
```

Comment:
The bug happened here, a `NoDelta` null delta marker got passed into this line and `ORDictionary` could not cast it to `DeltaGroup`